packages/app/src/elements/audit.ts

```diff
@@ -0,0 +1,239 @@
+
+                if (passwordFields.length === 0) {
+                    continue;
+                }
+
+                let isReused = false;
+                let isWeak = false;
+                // let isCompromised = false;
+
+                for (const passwordField of passwordFields) {
+                    const passwordHash = await sha1(passwordField.value);
+
+                    // Perform reused audit
+                    if (Object.keys(firstMatchPerPasswordHash).includes(passwordHash)) {
+                        isReused = true;
+
+                        // Don't add the same item twice to the list, if there are more than one reused password fields in it
+                        if (!reusedPasswordItemIds.has(item.id)) {
+                            reusedPasswords.push({ item, vault });
+                            reusedPasswordItemIds.add(item.id);
+                        }
+
+                        // TODO: Save audit match boolean in field
+                        // passwordField.auditResult = { ...(passwordField.auditResult || {}), isReused };
+
+                        // Also tag the first matching item as reused, once
+                        const firstMatch = firstMatchPerPasswordHash[passwordHash];
+                        if (!reusedPasswordItemIds.has(firstMatch.item.id)) {
+                            reusedPasswords.push({ item: firstMatch.item, vault: firstMatch.vault });
+                            reusedPasswordItemIds.add(firstMatch.item.id);
+
+                            // TODO: Save audit match boolean in field
+                            // firstMatch.passwordField.auditResult = { ...(firstMatch.passwordField.auditResult || {}), isReused };
+                        }
+                    }
+
+                    firstMatchPerPasswordHash[passwordHash] = { item, vault, passwordField };
+
+                    // Perform weak audit
+                    if (isPasswordWeak(passwordField.value)) {
+                        isWeak = true;
+
+                        // Don't add the same item twice to the list, if there are more than one reused password fields in it
+                        if (!weakPasswordItemIds.has(item.id)) {
+                            weakPasswords.push({ item, vault });
+                            weakPasswordItemIds.add(item.id);
+                        }
+
+                        // TODO: Save audit match boolean in field
+                        // passwordField.auditResult = { ...(passwordField.auditResult || {}), isWeak };
+                    }
+
+                    // TODO: Perform compromised audit
+                }
+
+                // TODO: Save audit match booleans in item
+                // item.auditResult = { lastAudited: new Date(), isReused, isWeak, isCompromised };
+
+                console.log({ item, isReused, isWeak });
+            }
+        }
+
+        // TODO: Sync?
```

Comment:
@MaKleSoft I haven't looked into if this is necessary after saving/updating the `auditResult` property in the items and fields, but let me know if there's any gotchas.

Reply:
This is enough for persisting the changes:
```ts
app.save();
```
If you want to manually trigger a sync, you can call:
```ts
app.synchronize();
```
But that's not really necessary, as the app will periodically sync everything automatically.